### PR TITLE
handling of lonepairs in the dummy region

### PR DESCRIPTION
**Pull request by JohannesKarwou**
## Description
It would be nice to be able to handle lonepairs which are in the dummy region

## Current problems with LPs:
-  The rdkit does not recognize lone pairs and thus they don't appear in the `common_core` list

## Ideas to solve this:
- parmed knows the atom type of lonepairs, mayber there is a way to create a function which checks if lonepairs are present and whether they are bound to a atom in the common core or in the dummy region

## First workaround:
 
One can define the variable `odered_connected_dummy_regions_cc1` in the jupyter-notebook, then for fhe `finish_common_core` one can use `s1_to_s2.finish_common_core(odered_connected_dummy_regions_cc1=odered_connected_dummy_regions_cc1)`

### CODE REVIEW

File: transformato/system.py [hunk 1]
```diff
@@ -235,22 +235,22 @@ def _initialize_system(
...
-                    frame_frame.append(atom.frame)
+            # g = psf.groups
+            # frame_idx = []
+            # frame_frame = []
+            # for atom in psf.atoms:
+            #     if hasattr(atom, "frame"):
+            #         frame_idx.append(atom.idx)
+            #         frame_frame.append(atom.frame)
 
             psf = psf[
```
Comment (JohannesKarwou) [inline transformato/system.py:246]: For rsfe the vacuum part is not working properly since no NUMLP section is included in the resulting psf

Comment (JohannesKarwou) [reply transformato/system.py:246]: @wiederm do you have an idea how to circumvent that, or should we just leave it like that and throw an error message if someone wants to do rsfe calculations with lonepairs?

Comment (wiederm) [reply transformato/system.py:246]: yes, I think we should through a `NonImplemented` error here.
In the long run I will try to fix this, but for now we can move forward with this I think.

**Comment by JohannesKarwou** [comment]
There are some additional `toppar` directories which we should reduce, but I suggest doing that in the [cleaned_testing](https://github.com/wiederm/transformato/tree/cleaned_testing) branch. I already opened a pull request, where I will collect all things we started discussing during the last months (https://github.com/wiederm/transformato/pull/54).

**Review by wiederm — APPROVED** [review]
thanks for addressing the comments!
This looks good!

File: transformato/mutate.py [hunk 2]
```diff
@@ -1013,18 +1082,23 @@ def _mutate_to_common_core(self, dummy_region: DummyRegion, cc_idx: list) -> dic
...
-        for atom in mol.GetAtoms():
-            idx = atom.GetIdx()
+
+        for atom in psf.view[f":{tlc}"].atoms:
+            # idx = atom.idx - self.offset
+            idx = atom.idx
             if idx not in cc_idx:
-                # hydrogens are collected seperatly IF they are not terminal dummy atoms
-                if atom.GetSymbol() == "H" and idx not in list_termin_dummy_atoms:
+                if atom.name.find("H") == False and idx not in list_termin_dummy_atoms:
```
[the remark this answers is not in the document]
Comment (JohannesKarwou) [reply transformato/mutate.py]: Oh your right! The problem is, that `mol.GetAtoms()` is related to a rdkit class and there are no lp available, so we can't use that.  I changed `atom.name.find("H") == False` to `if atom.type == "H"`, I hope that does the trick...

**Comment by JohannesKarwou** [comment]
In principle, that would be possible too, but I think that would make things even more complicated. Overall, the use of parmed seems quite okay for this task, since in the next parmed version there will be full support for lone pairs.

File: transformato/mutate.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (wiederm) [reply transformato/mutate.py:1102]: yes, I think t hat does the trick! Thanks!

**Pull request merged**